src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs

```
@@ -159,7 +157,7 @@ Language ID
                     // Default keyboard for language
                     string keyName = Convert.ToString(language, 16);
                     keyName = PadWithZeroes(keyName, 8);
-                    RegistryKey key = Registry.LocalMachine.OpenSubKey("SYSTEM\\CurrentControlSet\\Control\\Keyboard Layouts\\" + keyName);
+                    RegistryKey key = Registry.LocalMachine.OpenSubKey("SYSTEM\\CurrentControlSet\\Control\\Keyboard Layouts\\" + keyName)!;
```

this can obviously return `null` if the key doesn't exist. I wasn't sure what to do here, as this isn't expected and would indicate a corrupt (I think) registry.

There are a bunch of possible null dereferences around here. I don't know how to test them - I don't think we can mock `Registry.LocalMachine.OpenSubKey` sadly.

What do you think we should do?

---

I'm working on profiler API based mocking, then you'll be able to mock these things easily. I currently have a low level prototype and still need to build a high level API around it, will probably a few weeks until I have something publishable.

---

I'll make an issue

---

@JeremyKuhne any thoughts on how we could test this?

---

I don't think it helps us to ignore `!` things that clearly _can_ come back null. That is defeating the purpose of adding these annotations. There is bug here in that we aren't checking. We shouldn't fall over if a given key isn't in the registry, no matter how "normal" it is for it to be there. If we can't continue in any rational way we can throw deterministically, rather than null ref. In this case we should return null as we would if we can't find `"Layout Text"` below.

``` C#
RegistryKey? key = // ...
if (key == null)
{
    return null;
}

// Existing logic
```

I believe we should be extremely careful about introducing `!` and I would presume, by default, that they're bugs without detailed comments on why they are _really_ necessary.